### PR TITLE
Update network connection fields

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

The OTel semantic conventions for network connection type have evolved: `net.host.connection_type` is now `net.host.connection.type` and `net.host.connection.subtype`. See https://github.com/open-telemetry/opentelemetry-specification/pull/1759. This PR extends https://github.com/elastic/apm-server/pull/5436 to adapt to these changes.

Support Elastic APM agents sending "network.connection.type" ("n.c.t" in RUMv3) in metadata, which will be added to all events in the stream. This is currently intended for RUM. In the future we may want to support setting this at the even level too, to cater for operation-specific network information.

Finally, the Elasticsearch field names have been changed to `network.connection.type` and `network.connection.subtype`.

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)
~- [ ] Documentation has been updated~

## How to test these changes

1. Run the iOS agent
2. Check that the `network.connection.type` and `network.connection.subtype` fields are recorded

## Related issues

https://github.com/elastic/apm-server/issues/5203

### CODE REVIEW

File: model/error/_meta/fields.yml
Comment (simitt) [inline model/error/_meta/fields.yml:363]: Should we change to `connection.type` and `connection.subtype` on the document level?

**Review by simitt — COMMENTED** [review]
What is the urgency of this PR being backported to `7.14`? Does the iOS agent not work with the current fields, or would the `network.connection*` fields just be ignored?

File: model/error/_meta/fields.yml
Comment (bryce-b) [reply model/error/_meta/fields.yml:363]: I haven't made the `.type` & `.subtype` updates to the agent yet, since the spec hasn't been merged yet. So the iOS agent is still sending `net.host.connection_type` and not yet sending anything related to `subtype`.

Comment (simitt) [reply model/error/_meta/fields.yml:363]: In this case I suggest we keep what we have for `7.14` and keep this PR open until the spec is merged.

**Comment by bryce-b** [comment]
@simitt I mentioned to Andrew in slack: we don't need to backport this to 7.14; I've been holding off on updating the agent with these spec changes due to the Otel spec change not being merged yet.

File: model/error/_meta/fields.yml
Comment (axw) [reply model/error/_meta/fields.yml]: I've made this change now.

**Review by stuartnelson3 — APPROVED** [review]

**Comment by simitt** [comment]
Cannot yet be tested as the latest version of the iOS agent does not contain the change; it still sends `net.host.connection_type` instead of `net.host.connection.type`, see https://github.com/elastic/apm-agent-ios/blob/main/Sources/Instrumentation/NetworkInfo/NetworkStatusInjector.swift#L28.

@bryce-b any update on when this will be changed in the agent?

**Comment by bryce-b** [comment]
@Silvia These changes will be in this week.

**Pull request merged**
